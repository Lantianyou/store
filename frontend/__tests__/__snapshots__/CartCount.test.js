import { shallow, mount } from 'enzyme'
import toJSON from 'enzyme-to-json'
import CartItemCount from '../../components/CartItemCount'

describe('<CartItemCount />', () => {
  it('renders', () => {
    shallow(<CartItemCount count={10} />)
  })
  it('matches the count', () => {
    const wrapper = shallow(<CartItemCount count={11} />)
    expect(toJSON(wrapper)).toMatchSnapshot()
  })
  it('change the props', () => {
    const wrapper = shallow(<CartItemCount count={11} />)
    expect(toJSON(wrapper)).toMatchSnapshot()
    wrapper.setProps({ count: 90 })
    expect(toJSON(wrapper)).toMatchSnapshot()
  })
})
