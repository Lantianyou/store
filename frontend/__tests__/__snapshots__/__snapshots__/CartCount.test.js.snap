// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CartItemCount /> change the props 1`] = `
<CartItemCount__AnimationStyles>
  <TransitionGroup
    childFactory={[Function]}
    component="div"
  >
    <CSSTransition
      className="count"
      classNames="count"
      key="11"
      timeout={
        Object {
          "enter": 400,
          "exit": 400,
        }
      }
      unmountOnExit={true}
    >
      <CartItemCount__Dot>
        11
      </CartItemCount__Dot>
    </CSSTransition>
  </TransitionGroup>
</CartItemCount__AnimationStyles>
`;

exports[`<CartItemCount /> change the props 2`] = `
<CartItemCount__AnimationStyles>
  <TransitionGroup
    childFactory={[Function]}
    component="div"
  >
    <CSSTransition
      className="count"
      classNames="count"
      key="90"
      timeout={
        Object {
          "enter": 400,
          "exit": 400,
        }
      }
      unmountOnExit={true}
    >
      <CartItemCount__Dot>
        90
      </CartItemCount__Dot>
    </CSSTransition>
  </TransitionGroup>
</CartItemCount__AnimationStyles>
`;

exports[`<CartItemCount /> matches the count 1`] = `
<CartItemCount__AnimationStyles>
  <TransitionGroup
    childFactory={[Function]}
    component="div"
  >
    <CSSTransition
      className="count"
      classNames="count"
      key="11"
      timeout={
        Object {
          "enter": 400,
          "exit": 400,
        }
      }
      unmountOnExit={true}
    >
      <CartItemCount__Dot>
        11
      </CartItemCount__Dot>
    </CSSTransition>
  </TransitionGroup>
</CartItemCount__AnimationStyles>
`;
